Sources/Features/Shared/CloudKit/CloudKitError.swift

```diff
@@ -38,3 +52,148 @@ public struct CloudKitError: CloudKitErrorType {
         underlyingError = error
     }
 }
+
+
+
+// MARK: - Batch Modify Error Handling
+
+internal extension Array where Element: Equatable {
+
+    func bisect() -> (left: [Element], right: [Element]) {
+        let half = count / 2
+        return (Array(prefixUpTo(half)), Array(suffixFrom(half)))
+    }
+}
+
+internal extension BatchModifyOperationType where Save: Equatable, Save == Error.Save, Delete: Equatable, Delete == Error.Delete {
+
+    typealias ToModify = (toSave: [Save]?, toDelete: [Delete]?)
+    typealias ToModifyResponse = (left: ToModify, right: ToModify)
+
+    func bisect(error: Error) -> ToModifyResponse {
+        var left: ToModify = (.None, .None)
+        var right: ToModify = (.None, .None)
+
+        let remainingToSave = toSave?.filter { error.saved?.contains($0) ?? false }
+        let remainingToDelete = toDelete?.filter { error.deleted?.contains($0) ?? false }
```

Comment:
Condition should be something like: 
`let remainingToDelete = toDelete?.filter { !(error.deleted?.contains($0) ?? false) }`